From cd1421bcac1b2b444fa2e72869d42829c01290d4 Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Wed, 2 Jan 2019 16:25:29 -0800
Subject: [PATCH] TRITON-1065 cn-agent machine_destroy task does not use
 correct var for VM uuid in ensureProvisionComplete Reviewed by: Josh Wilsdon
 <josh@wilsdon.ca> Approved by: Josh Wilsdon <josh@wilsdon.ca>

---
 lib/backends/smartos/common.js                | 5 ++++-
 lib/backends/smartos/tasks/machine_destroy.js | 4 ++--
 package.json                                  | 2 +-
 3 files changed, 7 insertions(+), 4 deletions(-)

diff --git a/lib/backends/smartos/common.js b/lib/backends/smartos/common.js
index cb58295..cb9687b 100644
--- a/lib/backends/smartos/common.js
+++ b/lib/backends/smartos/common.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2018, Joyent, Inc.
+ * Copyright (c) 2019, Joyent, Inc.
  */
 
 /*
@@ -124,6 +124,9 @@ function provisionInProgressFile(uuidOrZonename, callback) {
 }
 
 function ensureProvisionComplete(uuid, callback) {
+    assert.uuid(uuid, 'uuid');
+    assert.func(callback, 'callback');
+
     var filename = '/var/tmp/machine-provision-' + uuid;
     var expiresAt;
     var timeoutMinutes = 10;
diff --git a/lib/backends/smartos/tasks/machine_destroy.js b/lib/backends/smartos/tasks/machine_destroy.js
index 83e2cb2..0d1650a 100644
--- a/lib/backends/smartos/tasks/machine_destroy.js
+++ b/lib/backends/smartos/tasks/machine_destroy.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2018, Joyent, Inc.
+ * Copyright (c) 2019, Joyent, Inc.
  */
 
 var common = require('../common');
@@ -44,7 +44,7 @@ function start() {
     vasync.pipeline({
         funcs: [
             function ensureProvisionComplete(_, cb) {
-                common.ensureProvisionComplete(self.req.uuid, cb);
+                common.ensureProvisionComplete(uuid, cb);
             },
             function detectLogadmBugWorkaroundRequired(_, cb) {
                 /*
diff --git a/package.json b/package.json
index f5d79ca..a7fd8d5 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "cn-agent",
   "description": "Triton Compute Node Agent",
-  "version": "2.6.1",
+  "version": "2.6.2",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
-- 
2.21.0

